pkg/resources/update_job.go

```diff
@@ -103,7 +103,9 @@ func PrepareMattermostJobTemplate(name, namespace string, baseDeployment *appsv1
 	// Override values for job-specific behavior.
 	job.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyNever
 	for i := range job.Spec.Template.Spec.Containers {
-		job.Spec.Template.Spec.Containers[i].Command = []string{"mattermost", "version"}
+		if job.Spec.Template.Spec.Containers[i].Name == "mattermost" {
+			job.Spec.Template.Spec.Containers[i].Command = []string{"mattermost", "version"}
+		}
 	}
```

Comment:
You should be able to use [this function](https://github.com/mattermost/mattermost-operator/blob/master/apis/mattermost/v1beta1/mattermost_utils.go#L152) to grab the `mattermost` container and likely remove the `for` loop entirely here since the intended target of this command override should just be the `mattermost` container.

Reply:
That wouldn't work out of the box because the container being returned is a copy. I can probably modify the function to pass a reference instead, but that might have other unintended side effects.

Reply:
Ah yep good point. Probably best to not try to modify that and just go with the way you approached it already. I'm interested in this fix as well because we're using istio injection (side cars).